Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/Instance/InstanceDomGenerator.cpp

```diff
@@ -126,6 +133,74 @@ namespace AzToolsFramework
             return true;
         }
 
+        bool InstanceDomGenerator::GenerateEntityDom(PrefabDom& entityDom, const AZ::Entity& entity) const
+        {
+            // Retrieves the focused instance.
+            auto prefabFocusInterface = AZ::Interface<PrefabFocusInterface>::Get();
+            if (!prefabFocusInterface)
+            {
+                AZ_Assert(false, "Prefab - InstanceDomGenerator::GenerateEntityDom - "
+                    "PrefabFocusInterface could not be found.");
+                return false;
+            }
+
+            InstanceOptionalReference focusedInstance = prefabFocusInterface->GetFocusedPrefabInstance(s_editorEntityContextId);
+            if (!focusedInstance.has_value())
+            {
+                AZ_Assert(false, "Prefab - InstanceDomGenerator::GenerateEntityDom - "
+                    "Could not get the focused instance. It should not be null.");
+                return false;
+            }
+
+            const AZ::EntityId entityId = entity.GetId();
+            InstanceOptionalReference owningInstance = m_instanceEntityMapperInterface->FindOwningInstance(entityId);
+            if (!owningInstance.has_value())
+            {
+                AZ_Assert(false, "Prefab - InstanceDomGenerator::GenerateEntityDom - "
+                    "Could not get the owning instance for the given entity id.");
+                return false;
+            }
+
+            // Climbs up from the owning instance to root instance, but stops at the focused instance if they can meet.
+            const InstanceClimbUpResult climbUpResult = PrefabInstanceUtils::ClimbUpToTargetOrRootInstance(
+                owningInstance->get(), &(focusedInstance->get()));
+            const Instance* focusedOrRootInstance = climbUpResult.m_reachedInstance;
+            if (!focusedOrRootInstance)
+            {
+                AZ_Assert(false, "Prefab - InstanceDomGenerator::GenerateEntityDom - "
+                    "Could not get the focused or root instance from climb-up. It should not be null.");
+                return false;
+            }
+
+            // Prepares the path from the focused (or root) instance to the entity.
+            AZStd::string relativePathFromTop = PrefabInstanceUtils::GetRelativePathFromClimbedInstances(climbUpResult.m_climbedInstances);
+            relativePathFromTop.append(m_instanceToTemplateInterface->GenerateEntityAliasPath(entityId));
+            PrefabDomPath relativeDomPathFromTopToEntity(relativePathFromTop.c_str());
+
+            // Retrieves the entity DOM from the template of the focused or root instance.
+            PrefabDom focusedOrRootTemplateDom;
+            focusedOrRootTemplateDom.CopyFrom(m_prefabSystemComponentInterface->FindTemplateDom(focusedOrRootInstance->GetTemplateId()),
```

Comment:
We can avoid the copy here if we directly use the template dom here. But before we do that, we would need to implement the previous suggestion to store the updated container entity dom in the focus handler itself. can you create a backlog task for this?

So the ideal flow would be:
1. if entity is container entity of focused prefab-> fetch it from the stored dom in focus handler
2. else, fetch it from the template directly

In either cases, you wouldn't need to do a copy.